authorization/service/deleteDevices.go

```diff
@@ -0,0 +1,70 @@
+package service
+
+import (
+	"context"
+
+	grpc_auth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"github.com/plgd-dev/cloud/authorization/pb"
+	"github.com/plgd-dev/cloud/pkg/log"
+	"github.com/plgd-dev/cloud/pkg/net/grpc"
+	"github.com/plgd-dev/kit/strings"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func getUniqueDeviceIds(deviceIds []string) []string {
+	devices := make(strings.Set)
+	devices.Add(deviceIds...)
+	delete(devices, "")
+	return devices.ToSlice()
+}
+
+// DeleteDevices removes a devices from user.
+func (s *Service) DeleteDevices(ctx context.Context, request *pb.DeleteDevicesRequest) (*pb.DeleteDevicesResponse, error) {
+	tx := s.persistence.NewTransaction(ctx)
+	defer tx.Close()
+
+	owner := request.UserId
+	if owner == "" {
+		if token, err := grpc_auth.AuthFromMD(ctx, "bearer"); err == nil {
+			uid, err := grpc.ParseOwnerFromJwtToken(s.ownerClaim, token)
+			if err == nil {
+				owner = uid
+			}
+		}
+	}
+
+	if owner == "" {
+		return nil, log.LogAndReturnError(status.Errorf(codes.InvalidArgument, "cannot delete devices: invalid UserId"))
+	}
+
+	deviceIds := getUniqueDeviceIds(request.DeviceIds)
+	if len(deviceIds) == 0 {
+		return nil, log.LogAndReturnError(status.Errorf(codes.InvalidArgument, "cannot delete devices: invalid DeviceIds"))
+	}
+
+	// TODO validate jwt token -> only jwt token is supported
+
+	var deletedDeviceIds []string
+	for _, deviceId := range request.DeviceIds {
```

Comment:
if deviceIDs are empty you don't proceed any deletein auth service. but in ra you delete all resources of user.